#include "mainwindow.h"
#include "ui_mainwindow.h"

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent),
      numberDivisions(25)
    , ui(new Ui::MainWindow)
{
    ui->setupUi(this);
        //QObject::connect(ui->calc_button, &QPushButton::clicked, this, &MainWindow::calc_button_clicked);
}

MainWindow::~MainWindow()
{
    delete ui;
}


void MainWindow::on_calc_button_clicked(bool checked)
{
    // ???????? ??????????? ?? ????
    double coef_a = ui->coef_a->text().toDouble();
    double coef_b = ui->coef_b->text().toDouble();
    double coef_c = ui->coef_c->text().toDouble();
    //????????? ??? ?????
    QVector<QPointF> points;
    //???
    qreal step = 0.05;
    //??????????? ????? ??????? ?? ???? ? ????????????? ??????? ??????? ?? ?
    for (qreal p = 0.; p >= -numberDivisions / 2; p -= step)
        points.push_back(QPointF(p, (coef_a*p*p) + (coef_b * p) + coef_c));
    // ?????????? ?????????, ?????? ?????? ???????? ??????? ???????? ??????
    std::reverse(points.begin(), points.end());
    //??????????? ????? ??????? ?? ??????? ???? ? ????????????? ??????? ??????? ?? ?
    for (qreal p = step; p <= numberDivisions / 2; p += step)
        points.push_back(QPointF(p, (coef_a*p*p) + (coef_b * p) + coef_c));

    ui->render_area->setAxes(renderArea::Axis("x", numberDivisions), renderArea::Axis("y", numberDivisions));

    //???????? ????? ? ??????? ?????????
    ui->render_area->setGraphPoints(points);
}

